Work with regular expression (#3024)
influxdata/docs-v2: content/flux/v0.x/data-types/regexp.md

seotitle: Work with Flux regular expression types
list_title: Regular expression types
description: >
  Learn how to work with Flux regular expression types.
menu:
  flux_0_x:
    name: Regular expression types
    parent: Work with data types
weight: 101
flux/v0.x/tags: ["regexp types", "data types"]
related:
  - /flux/v0.x/stdlib/regexp/
---

A **regular expression** type represents a regular expression pattern.

**Type name**: `regexp`

###### On this page
- [Regular expression syntax](#regular-expression-syntax)
- [Use regular expression flags](#use-regular-expression-flags)
- [Use regular expressions in predicate expressions](#use-regular-expressions-in-predicate-expressions)
- [Convert a string to a regular expression](#convert-a-string-to-a-regular-expression)
- [Examples](#examples)

## Regular expression syntax

> Flux uses the [Go regexp implementation and syntax](https://pkg.go.dev/regexp)
> which uses the same _general_ syntax as Perl, Python, and other languages.
> Regular expression literals are enclosed in forward slash characters (`/`).

Reviewer: ```suggestion
Flux uses the [Go regexp implementation and syntax](https://pkg.go.dev/regexp).
This syntax is similar to the regular expressions Perl, Python, and other languages.
Regular expression literals are enclosed in forward slash characters (`/`).
```

Reviewer 2: ```suggestion
Flux uses the [Go regexp implementation and syntax](https://pkg.go.dev/regexp).
This syntax is similar to regular expressions in Perl, Python, and other languages.
Regular expression literals are enclosed in forward slash characters (`/`).
```

- [Escape regular expression metacharacters in a string](#escape-regular-expression-metacharacters-in-a-string)

### Replace all substrings that match a regular expression
1. Import the [`regexp` package](/flux/v0.x/stdlib/regexp).
2. Use [`regexp.replaceAllString()`](/flux/v0.x/stdlib/regexp/replaceallstring/)
   and provide the following parameters:

    - **r**: regular expression
    - **v**: string to search
    - **t**: replacement for matches to **r**.

```js
import "regexp"

regexp.replaceAllString(
  r: /a(x*)b/,
  v: "-ab-axxb-",
  t: "T"
)
// Returns "-T-T-"
```

### Return the first regular expression match in a string
1. Import the [`regexp` package](/flux/v0.x/stdlib/regexp).
2. Use [`regexp.findString()`](/flux/v0.x/stdlib/regexp/findstring/) to return
   the first regular expression match in a string.
   Provide the following parameters:

    - **r**: regular expression
    - **v**: string to search  

```js
import "regexp"

regexp.findString(
  r: /foo.?/,
  v: "seafood fool"
)
// Returns "food"
```

### Escape regular expression metacharacters in a string

> 1. Import the [`regexp` package](/flux/v0.x/stdlib/regexp).

Reviewer: Maybe add a bit for context here?

```suggestion
Certain characters must be escaped in order to use them as literals in regular expressions.
This can make the expression hard to read.
Flux provides a convenience method `regex.quoteMeta()`.

1. Import the [`regexp` package](/flux/v0.x/stdlib/regexp).
```
